### PR TITLE
Fix error handling in disjoint_pool_initialize()

### PR DESCRIPTION
**Pull request by ldorau**

### CODE REVIEW

**Comment by ldorau** [comment]
@igchor Those are fixes from your review: https://github.com/oneapi-src/unified-memory-framework/pull/4

**Review by vinser52 — APPROVED** [review]

File: src/pool/disjoint/pool_disjoint.cpp [hunk 1]
```diff
@@ -29,22 +29,15 @@ disjoint_pool_initialize(umf_memory_provider_handle_t provider,
...
 
-    struct disjoint_memory_pool *pool_data =
-        (struct disjoint_memory_pool *)malloc(
-            sizeof(struct disjoint_memory_pool));
-    if (!pool_data) {
-        fprintf(stderr, "cannot allocate memory for metadata\n");
-        abort();
-    }
-
+    struct disjoint_memory_pool *pool_data = new struct disjoint_memory_pool;
```
Comment (igchor) [inline src/pool/disjoint/pool_disjoint.cpp:32]: You should also update the finalize function (free -> delete).

**Comment by igchor** [comment]
> @igchor Those are fixes from your review: #4

@ldorau thanks! Just one comment (above).

File: src/pool/disjoint/pool_disjoint.cpp (hunk 1 above)
Comment (ldorau) [reply src/pool/disjoint/pool_disjoint.cpp:32]: Right, I have forgotten about that. Done:
https://github.com/oneapi-src/unified-memory-framework/pull/19
Thanks!

**Comment by ldorau** [comment]
> @ldorau thanks! Just one comment (above).

@igchor  Done: https://github.com/oneapi-src/unified-memory-framework/pull/19

**Pull request merged**